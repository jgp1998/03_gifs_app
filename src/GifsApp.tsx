
import { GiftList } from "./gifs/components/GiftList";
import { PreviousSearch } from "./gifs/components/PreviousSearch";
import { useGifs } from "./gifs/hooks/useGifs";
import { CustomHeader } from "./shared/components/CustomHeader";
import { SearchBar } from "./shared/components/SearchBar";
export const GifsApp = () => {
    const {
        handleSearch,
        previousTerms,
        handleTermClicked,
        gifs
    } = useGifs();


    return (
        <>
            <CustomHeader
                title="Buscador de Gifs"
                description="Encuentra los mejores gifs aquÃ­"
            />
            {/* SearchBar */}
            <SearchBar
                placeholder="Busca un gif..."
                btnText="Buscar"
                onQuery={handleSearch}
            />
            {/* Busquedas previas */}
            <PreviousSearch
                onLabelClicked={(term) => handleTermClicked(term)}
                subtitle="Busquedas Previas"
                searches={previousTerms}
            />
            <GiftList
                gifs={gifs}
            />
        </>
    )
}
