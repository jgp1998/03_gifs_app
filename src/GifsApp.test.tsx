import { describe, expect, it } from 'vitest';
import { render } from '@testing-library/react';
import { GifsApp } from './GifsApp';

describe('GifsApp', () => {
    it('should render correctly', () => {
        const { container } = render(<GifsApp />);
        expect(container).toMatchSnapshot();

    });
});
