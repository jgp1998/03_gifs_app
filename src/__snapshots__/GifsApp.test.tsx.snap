// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`GifsApp > should render correctly 1`] = `
<div>
  <div
    class="content-center"
  >
    <h1>
      Buscador de Gifs
    </h1>
    <p>
      Encuentra los mejores gifs aqu√≠
    </p>
  </div>
  <div
    class="search-container"
  >
    <input
      placeholder="Busca un gif..."
      type="text"
      value=""
    />
    <button>
      Buscar
    </button>
  </div>
  <div
    class="previous-searches"
  >
    <h2>
      Busquedas Previas
    </h2>
    <ul
      class="previous-searches-list"
    />
  </div>
  <div
    class="gifs-container"
  />
</div>
`;
